File: conversions/django_models/family.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Family(AbstractElementModel):

    # Community
    spirit = models.TextField(blank=True, null=True)
    alliances = models.ManyToManyField("Family", blank=True, related_name="family_alliances")
    rivalries = models.ManyToManyField("Family", blank=True, related_name="family_rivalries")

    # Lineage
    history = models.TextField(blank=True, null=True)
    ancestors = models.ManyToManyField("Character", blank=True, related_name="family_ancestors")
    traits = models.ManyToManyField("Trait", blank=True, related_name="family_traits")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="family_abilities")
    languages = models.ManyToManyField("Language", blank=True, related_name="family_languages")

    # World
    status = models.TextField(blank=True, null=True)
    competition = models.ManyToManyField("Institution", blank=True, related_name="family_competition")
    administrates = models.ManyToManyField("Institution", blank=True, related_name="family_administrates")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="family_creatures")

    # Legacy
    traditions = models.TextField(blank=True, null=True)
    estate = models.ForeignKey("Location", on_delete=models.SET_NULL, blank=True, null=True, related_name="family_estate")
    heirlooms = models.ManyToManyField("Object", blank=True, related_name="family_heirlooms")

    def __str__(self):
        return self.name

File: conversions/csharp_unity_json_properties/Family.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Family : Element
{
    [JsonProperty("spirit"), TextAttribute("")]
    public string spirit;
    [JsonProperty("alliances"), ReferenceAttribute(typeof(Family), true)]
    public string alliances;
    [JsonProperty("rivalries"), ReferenceAttribute(typeof(Family), true)]
    public string rivalries;
    [JsonProperty("history"), TextAttribute("")]
    public string history;
    [JsonProperty("ancestors"), ReferenceAttribute(typeof(Character), true)]
    public string ancestors;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("status"), TextAttribute("")]
    public string status;
    [JsonProperty("competition"), ReferenceAttribute(typeof(Institution), true)]
    public string competition;
    [JsonProperty("administrates"), ReferenceAttribute(typeof(Institution), true)]
    public string administrates;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("traditions"), TextAttribute("")]
    public string traditions;
    [JsonProperty("estate"), ReferenceAttribute(typeof(Location))]
    public string estate;
    [JsonProperty("heirlooms"), ReferenceAttribute(typeof(Object), true)]
    public string heirlooms;
}
